src/App.tsx

```diff
@@ -1,61 +1,106 @@
 import './App.scss';
+import { FormEvent, useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userName, setUserName] = useState('');
+  const [isTitleEmpty, setIsTitleEmpty] = useState(false);
+  const [isUserSelected, setIsUserSelected] = useState(false);
+
+  const [todos, setTodos] = useState(todosFromServer.map(todo => {
+    const user = usersFromServer
+      .find(serverUser => serverUser.id === todo.userId) || null;
+
+    return { ...todo, user };
+  }));
+
+  const handleTitleChange = (event: FormEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    setIsTitleEmpty(false);
+  };
+
+  const handleUserChange = (event: FormEvent<HTMLSelectElement>) => {
+    setUserName(event.currentTarget.value);
+    setIsUserSelected(false);
+  };
+
+  const addTodo = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const selectedUser = usersFromServer.find(user => user.name === userName);
+    const idArr = todos.map(todo => todo.id);
+    const maxId = Math.max(...idArr);
+
+    if (title.length < 1) {
+      setIsTitleEmpty(true);
+      if (!selectedUser) {
+        setIsUserSelected(true);
+      }
+    } else if (!selectedUser) {
+      setIsUserSelected(true);
```

Comment:
barely readable code structure, nested conditions are evil 🔪 
follow this approach
```
if // no title
  show error
  return

if // no user
  show error
  return

// then everything else
```